iphone/CoreApi/CoreApi/PlacePageData/ElevationProfile/ElevationProfileData.mm

```diff
@@ -20,21 +22,27 @@ @implementation ElevationProfileData
 
 @implementation ElevationProfileData (Core)
 
-- (instancetype)initWithElevationInfo:(ElevationInfo const &)elevationInfo
-                          activePoint:(double)activePoint
-                           myPosition:(double)myPosition {
+- (instancetype)initWithTrackId:(MWMTrackID)trackId
+                  elevationInfo:(ElevationInfo const &)elevationInfo
+                    activePoint:(double)activePoint
+                     myPosition:(double)myPosition {
   self = [super init];
   if (self) {
+    _trackId = trackId;
     _ascent = elevationInfo.GetAscent();
     _descent = elevationInfo.GetDescent();
     _maxAttitude = elevationInfo.GetMaxAltitude();
     _minAttitude = elevationInfo.GetMinAltitude();
     _difficulty = convertDifficulty(elevationInfo.GetDifficulty());
-    NSMutableArray *pointsArray = [NSMutableArray array];
-    for (auto const &p : elevationInfo.GetPoints()) {
-      ElevationHeightPoint *point = [[ElevationHeightPoint alloc] initWithDistance:p.m_distance
-                                                                       andAltitude:p.m_point.GetAltitude()];
-      [pointsArray addObject:point];
+
+    NSMutableArray * pointsArray = [NSMutableArray array];
+    for (auto const & point : elevationInfo.GetPoints()) {
+      auto pointLatLon = mercator::ToLatLon(point.m_point.GetPoint());
+      CLLocationCoordinate2D coordinates = CLLocationCoordinate2DMake(pointLatLon.m_lat, pointLatLon.m_lon);
+      ElevationHeightPoint * elevationPoint = [[ElevationHeightPoint alloc] initWithCoordinates:coordinates
+                                                                                       distance:point.m_distance
+                                                                                    andAltitude:point.m_point.GetAltitude()];
+      [pointsArray addObject:elevationPoint];
     }
     _points = [pointsArray copy];
```

Comment:
This `copy` method doesn't copy the points itself but return the _non mutable array_ as a pointer. 
In the other case the _mutable instance_ will be assigned to the instance instead of the _imutable_ as defined in the property signature that potentially may lead to the errors.